src/containers/Accounts/AMM/AMMAccounts/index.tsx

```diff
@@ -0,0 +1,140 @@
+import React, { useContext, useEffect, useState } from 'react'
+import { useParams, useRouteMatch } from 'react-router'
+import { useTranslation } from 'react-i18next'
+import { useLanguage } from '../../../shared/hooks'
+import '../../styles.scss'
+import formatBalance from '../../../../rippled/lib/txSummary/formatAmount'
+import {
+  getAccountTransactions,
+  getAMMInfo,
+} from '../../../../rippled/lib/rippled'
+import { Tabs } from '../../../shared/components/Tabs'
+import { analytics, ANALYTIC_TYPES, formatAsset } from '../../../shared/utils'
+import SocketContext from '../../../shared/SocketContext'
+import { ERROR_MESSAGES } from '../../Errors'
+import NoMatch from '../../../NoMatch'
+import {
+  AMMAccountHeader,
+  AmmDataType,
+} from './AMMAccountHeader/AMMAccountHeader'
+import { AccountTransactionTable } from '../../AccountTransactionTable'
+
+const getErrorMessage = (error: string) =>
+  ERROR_MESSAGES[error] || ERROR_MESSAGES.default
+
+function findAMMCreate(txs: [any]) {
+  const ammCreate = txs.filter((tx) => tx.tx.TransactionType === 'AMMCreate')
+
+  if (ammCreate.length < 1) throw new Error('Could not find AMM Create')
+
+  return ammCreate[0].tx
+}
+
+export const AMMAccounts = () => {
+  const { id: accountId, tab = 'transactions' } = useParams<{
+    id: string
+    tab: string
+  }>()
+  const { path = '/' } = useRouteMatch()
+  const { t } = useTranslation()
+  const mainPath = `${path.split('/:')[0]}/${accountId}`
+  const rippledSocket = useContext(SocketContext)
+  const [data, setData] = useState<AmmDataType>()
+  const [error, setError] = useState<any>()
+  const language = useLanguage()
+
+  useEffect(() => {
```

Comment:
This should use `useQuery`

Reply:
updated.

FYI: By using useQuery we lose dependency updating functionality. I.e. when language changes (one of the dependencies of this hook) we wont re-render the page with the new language. Updates happen when the cached data becomes stale (i think the default value is 1minute). 

Maybe im mistaken but I cant find documentation that says we can pass in a dependency list.

Reply:
Update: It seems like useQuery uses the query keys like useEffect dependencies: https://codemachine.dev/things-i-learned-while-using-react-query-part-2#treat-query-keys-like-a-dependency-array-useeffect-style

So the above FYI may not be relevant.